src/vs/workbench/services/preferences/common/preferencesValidation.ts

```diff
@@ -269,3 +285,51 @@ function getArrayOfStringValidator(prop: IConfigurationPropertySchema): ((value:
 
 	return null;
 }
+
+function getObjectValidator(prop: IConfigurationPropertySchema): ((value: any) => (string | null)) | null {
+	if (prop.type === 'object' && !prop.additionalProperties) {
+		const { properties, patternProperties } = prop;
+		return value => {
+			if (!value) {
+				return null;
+			}
+
+			const errors: string[] = [];
+
+			Object.keys(value).forEach((key: string) => {
+				const data = value[key];
+				if (properties && key in properties) {
+					const errorMessage = getErrorsForSchema(properties[key], data);
+					if (errorMessage) {
+						errors.push(nls.localize('validations.errorsForKey', 'For key "{0}": {1}\n', key, errorMessage));
```

Comment:
I would skip the words "For key". Just like `"some.setting": <error>`

Reply:
And don't put a `\n` at the end of a localized string like that, localizers might drop it, add that outside of the localized string.